Tests/LibDNS/TestDNSResolver.cpp

```diff
@@ -14,27 +14,45 @@ TEST_CASE(test_udp)
     Core::EventLoop loop;
 
     DNS::Resolver resolver {
-        [&] -> ErrorOr<DNS::Resolver::SocketResult> {
-            Core::SocketAddress addr = { IPv4Address::from_string("1.1.1.1"sv).value(), static_cast<u16>(53) };
-            return DNS::Resolver::SocketResult {
-                TRY(Core::BufferedSocket<Core::UDPSocket>::create(TRY(Core::UDPSocket::connect(addr)))),
-                DNS::Resolver::ConnectionMode::UDP,
+        [] -> NonnullRefPtr<Core::Promise<DNS::Resolver::SocketResult>> {
+            auto promise = Core::Promise<DNS::Resolver::SocketResult>::construct();
+
+            auto make_resolver = [] -> ErrorOr<DNS::Resolver::SocketResult> {
+                Core::SocketAddress addr = { IPv4Address::from_string("1.1.1.1"sv).value(), static_cast<u16>(53) };
+                return DNS::Resolver::SocketResult {
+                    TRY(Core::BufferedSocket<Core::UDPSocket>::create(TRY(Core::UDPSocket::connect(addr)))),
+                    DNS::Resolver::ConnectionMode::UDP,
+                };
             };
+
+            auto result = make_resolver();
+            if (!result.is_error()) {
+                promise->resolve(result.release_value());
+            } else {
+                promise->reject(result.release_error());
+            }
+
+            return promise;
         }
     };
 
-    TRY_OR_FAIL(resolver.when_socket_ready()->await());
+    auto when_socket_ready_promise = resolver.when_socket_ready();
```

Comment:
`socket_ready`/`socket_readiness` or `socket_is_ready`? (`when_` is the resolution part; `_promise` is unnecessary in typed environment? same for names below?)